### PR TITLE
[REVIEW] Updated document for XGBoost usage with dask-cuda. 

### LINKED ISSUE #5375: [DOC]  Update the dask-xgboost document to point to now updated version of xgboost usage.
## Report incorrect documentation

**Location of incorrect documentation**
Link to the current document: https://docs.rapids.ai/api/cudf/stable/dask-xgb-10min.html

**Describe the problems or issues found in the documentation**
The current document uses older method of using xgboost with dask [dask-xgboost](https://github.com/rapidsai/dask-xgboost/) but we have moved to using xgboost support API for dask [xgboost.dask](https://xgboost.readthedocs.io/en/latest/python/python_api.html?highlight=dask#module-xgboost.dask)

**Steps taken to verify documentation is incorrect**
* Verified that the current document shows implementation of xgboost using dask-xgboost. 
* Verified that the current document also has links pointing to the dask-xgboost repo.

**Suggested fix for documentation**
* Update the link of the document to point to xgboost dask API.
* Update the notebook to show example with using xgboost.dask instead of dask-xgboost.
* Update the heading of the document to be specific to this new change.
---

## Report needed documentation

**Report needed documentation**

**Describe the documentation you'd like**

**Steps taken to search for needed documentation**

### PR DESCRIPTION
**Pull request by think-high**
closes #5375

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: docs/cudf/source/dask-xgb-10min.ipynb [hunk 1]
```diff
@@ -92,75 +134,113 @@
...
    "outputs": [],
    "source": [
     "size = 1000000\n",
     "npartitions = 8\n",
     "\n",
-    "pdf = pd.DataFrame({'x': np.random.randint(0, npartitions, size=size), 'y': np.random.normal(size=size)})\n",
-    "pdf = dask.dataframe.from_pandas(pdf, npartitions=npartitions)\n",
-    "\n",
-    "ddf = dask_cudf.from_dask_dataframe(pdf)\n",
+    "cdf = cudf.DataFrame({'x': np.random.randint(0, npartitions, size=size), 'y': np.random.normal(size=size)})\n",
```
Comment (galipremsagar) [inline docs/cudf/source/dask-xgb-10min.ipynb:144]: We can use `cupy.random.randint` & `cupy.random.normal` to avoid H2D copies.

Comment (think-high) [reply docs/cudf/source/dask-xgb-10min.ipynb:144]: perfect. Will update.

File: docs/cudf/source/dask-xgb-10min.ipynb [hunk 2]
```diff
@@ -174,56 +265,109 @@
...
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "rmse value: 2.291584656127447\n"
+     ]
+    }
+   ],
+   "source": [
+    "rmse = np.sqrt(squared_error.mean().compute())\n",
```
Comment (galipremsagar) [inline docs/cudf/source/dask-xgb-10min.ipynb:368]: We could probably get rid of `numpy` usage altogether in this entire notebook if you use `cupy.sqrt` here

https://docs-cupy.chainer.org/en/stable/reference/generated/cupy.sqrt.html

**Comment by galipremsagar** [comment]
@think-high Could you add a changelog entry for CI to run?

File: docs/cudf/source/dask-xgb-10min.ipynb [hunk 3]
```diff
@@ -174,56 +270,144 @@
...
+    "1. `client`: the `dask.distributed.Client`\n",
+    "2. `booster`: the Booster produced by the XGBoost training session\n",
+    "3. `X_test`: an instance of `dask_cudf.core.DataFrame` containing the data to be inferenced (acquire predictions)\n",
     "\n",
-    "Finally, the mean can be computed by using an aggregator from the `dask_cudf` API. The entire computation is initiated via `.compute()`. We take the square-root of the result, leaving us with `rmse = np.sqrt(test.squared_error.mean().compute())`. Note: `.squared_error` is an accessor for `test[squared_error]`... Like so:"
+    "`pred` and `true` are instances of `dask_cudf.core.Series` object. We align the index of both the Series by resetting it through `reset_index`. To compute the root mean squared error(RMSE) we first compute difference squared between the `pred` and `true` series through the operation `squared_error = ((pred - true)**2)`\n",
+    "\n",
+    "Finally, the mean is computed by using an aggregator from the `dask_cudf` API. The entire computation is initiated via `.compute()`. We take the square-root of the result, leaving us with `rmse = np.sqrt(squared_error.mean().compute())`."
```
Comment (kkraus14) [inline docs/cudf/source/dask-xgb-10min.ipynb:364]:
```suggestion
    "Finally, the mean is computed by using an aggregator from the `dask_cudf` API. The entire computation is initiated via `.compute()`. We take the square-root of the result, leaving us with `rmse = cupy.sqrt(squared_error.mean().compute())`."
```

**Comment by kkraus14** [comment]
@galipremsagar could you review again when you get a chance?

**Comment by think-high** [comment]
Thanks. My bad, I was supposed to add the changelog entry but thanks @kkraus14 for taking it forward.

**Pull request merged**